lib/Runtime/Language/InterpreterStackFrame.cpp

```diff
@@ -6744,7 +6745,10 @@ const byte * InterpreterStackFrame::OP_ProfiledLoopBodyStart(const byte * ip)
             // mark the stackFrame as 'in try block'
             this->m_flags |= InterpreterStackFrameFlags_WithinTryBlock;
 
-            CacheSp();
+            if (shouldCacheSP)
```

Comment:
I see. Yes, we need ignore cache SP for both ProcessTryFinally for your second test, but how about if there are some more ProcessTryFinally following  but which are not for the previous yield? Seems this fix will also ignore caching SP for them.

---
In reply to: [76500192](https://github.com/Microsoft/ChakraCore/pull/1502#discussion_r76500192) [](ancestors = 76500192)